### PR TITLE
machine: Add `ConnectionInfo` to inspect

### PR DESCRIPTION
**Pull request by jakecorrenti**
Adds `ConnectionInfo()` to the `MachineConfig` and fills out `InspectInfo` accordingly. Additionally fixes the "inspect with go format" test.

#### Does this PR introduce a user-facing change?

```release-note
`ConnectionInfo` is now properly displayed in `machine inspect`
```

### CODE REVIEW

File: cmd/podman/machine/inspect.go [hunk 1]
```diff
@@ -72,12 +72,19 @@ func inspect(cmd *cobra.Command, args []string) error {
 			return err
 		}
 
+		podmanSocket, podmanPipe, err := provider.ConnectionInfo(name)
+		if err != nil {
+			return err
+		}
+
 		ii := machine.InspectInfo{
 			// TODO I dont think this is useful
```
Comment (vrothberg) [inline cmd/podman/machine/inspect.go:81]: Drive-by comment: Podman 5.0 seems like the right time to address (or remove) the TODO.

File: pkg/machine/applehv/stubber.go [hunk 2]
```diff
@@ -346,3 +347,14 @@ func (a AppleHVStubber) PostStartNetworking(mc *vmconfigs.MachineConfig, noInfo
 func (a AppleHVStubber) GetDisk(userInputPath string, dirs *define.MachineDirs, mc *vmconfigs.MachineConfig) error {
 	return diskpull.GetDisk(userInputPath, dirs, mc.ImagePath, a.VMType(), mc.Name)
 }
+
+func (a AppleHVStubber) ConnectionInfo(_ string) (*define.VMFile, *define.VMFile, error) {
+	sockName := "podman.sock"
```
Comment (vrothberg) [inline pkg/machine/applehv/stubber.go:352]: This smells like a code duplicate.  I didn't check but is there no other function to get the path to the socket?

File: cmd/podman/machine/inspect.go (hunk 1 above)
Comment (mheon) [reply cmd/podman/machine/inspect.go:81]: We agreed to change that to point at the actual config file, not the directory

Comment (baude) [reply cmd/podman/machine/inspect.go:81]: my memory was that we agreed to change the label so that it reflects Dir

File: pkg/machine/hyperv/stubber.go [hunk 3]
```diff
@@ -547,3 +547,16 @@ func createNetworkUnit(netPort uint64) (string, error) {
 	netUnit.Add("Install", "WantedBy", "multi-user.target")
 	return netUnit.ToString()
 }
+
+func (h HyperVStubber) ConnectionInfo(name string) (*define.VMFile, *define.VMFile, error) {
```
Comment (baude) [inline pkg/machine/hyperv/stubber.go:551]: would it make sense to hand this off of MachineConfig ?

Comment (jakecorrenti) [reply pkg/machine/hyperv/stubber.go:551]: Yes, I think that would be better. It would address the code duplication that @vrothberg mentioned above.

File: pkg/machine/applehv/stubber.go (hunk 2 above)
Comment (jakecorrenti) [reply pkg/machine/applehv/stubber.go:352]: There used to be `<provider>.forwardSocketPath()` but it looks like that got nuked with other podman 4 code. The duplication will be solved by moving the `ConnectionInfo` function to `MachineConfig` instead

File: cmd/podman/machine/inspect.go (hunk 1 above)
Comment (jakecorrenti) [reply cmd/podman/machine/inspect.go]: Fixed

**Comment by baude** [comment]
LGTM

**Comment by TomSweeneyRedHat** [comment]
LGTM
and happy green test buttons.
